### PR TITLE
feat(events): `EventBus.grantPutEventsTo` method for granular grants

### LINKED ISSUE #11228: [events] Grant events:PutEvents to specific EventBridge bus
Right now EventBus has static method grantPutEvents() which grants access to all EventBridge buses in account.
It would be useful to add same method or move it to instance method so it could grant access to send events only to the bus which method is called.

### Use Case

This would allow to manage grants more granular

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by DaWyz**
Right now EventBus has a static method `grantPutEvents()` which grants PutEvents to all EventBridge buses in the account.

Adding a `grantPutEventsTo()` method to the IEventBus interface that grants PutEvents to the specific event bus.
We are also deprecating `grantPutEvents()` in favor to `grantAllPutEvents()` which has the same behavior.

Closes #11228.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-events/lib/event-bus.ts [hunk 1]
```diff
@@ -47,6 +47,14 @@ export interface IEventBus extends IResource {
...
    */
   archive(id: string, props: BaseArchiveProps): Archive;
+
+  /**
+   * Grants an IAM Principal to send custom events to the eventBus
+   * so that they can be matched to rules.
+   *
+   * @param grantee The principal (no-op if undefined)
+   */
+  grantPutEventsTo(grantee: iam.IGrantable): iam.Grant;
```
Comment (DaWyz) [inline packages/@aws-cdk/aws-events/lib/event-bus.ts:57]: `grantPutEvents` is already taken by the static method. Should I drop the static method and rename this ? (That would be a breaking change..)

Comment (DaWyz) [reply packages/@aws-cdk/aws-events/lib/event-bus.ts:57]: Resolving as this a stable module and the contributing guide says no breaking changes on stable modules.

Comment (rix0rrr) [reply packages/@aws-cdk/aws-events/lib/event-bus.ts:57]: Yap, that is unfortunate.

Can I ask you to, in this same commit, to `@deprecate` the current `grantPutEvents` and add a new one that's called `grantAllPutEvents` as well? 

Then we are set up to rename this to `grantPutEvents` at some point in the future once we've removed deprecated APIs.

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-events/lib/event-bus.ts (hunk 1 above)
Comment (DaWyz) [reply packages/@aws-cdk/aws-events/lib/event-bus.ts:57]: Makes sense. It's done ! Ready for review

**Pull request merged**